srv/components/system/config/service_user.sls

```diff
@@ -36,7 +37,7 @@ service_user_configured:
     - shell: {{ user_data['shell'] }}
     - groups: {{ user_data['groups'] }}
     # would be activated at unboxing time
-    {% if not "primary" in pillar["cluster"][grains["id"]]["roles"] -%}
+    {% if roles and not "primary" in pillar["cluster"][grains["id"]]["roles"] -%}
```

Comment:
why do you resolve the roles from the pillar again here? could it be checked just from `roles` variable?

Reply:
comment is addressed